### PR TITLE
Third party validation payment

### PR DESCRIPTION
**Pull request by ankitdas13**
Doc reference :  https://razorpay.com/docs/payments/third-party-validation

### CODE REVIEW

File: documents/payment.md [hunk 1]
```diff
@@ -466,6 +466,191 @@ body, err := client.Payment.CreatePaymentJson(para_attr, nil)
...
+**Parameters:**
+| Name        | Type    | Description                          |
+|-------------|---------|--------------------------------------|
+| amount*          | integer | Amount of the order to be paid  |
+| currency*   | string  | The currency of the payment (defaults to INR)                                  |
+| order_id*        | string  | The unique identifier of the order created. |
+| email*        | string      | Email of the customer                       |
+| contact*      | string      | Contact number of the customer              |
+| method*      | string  | Possible value is `netbanking` |
+| bank      | string      | The customer's bank code.For example, `HDFC`.|
```
Comment (sofiya02) [inline documents/payment.md:493]: bank is mandatory

File: documents/payment.md [hunk 2]
```diff
@@ -466,6 +466,191 @@ body, err := client.Payment.CreatePaymentJson(para_attr, nil)
...
+| upi* (for Upi only)  | array      | All keys listed [here](https://razorpay.com/docs/payments/third-party-validation/s2s-integration/upi/collect#step-14-initiate-a-payment) are supported  |
+
+**Response:** <br>
+```json
+{
+  "razorpay_payment_id": "pay_EAm09NKReXi2e0"
+}
+```
+-------------------------------------------------------------------------------------------------------
+### Create Payment UPI s2s / VPA token (Third party validation)
```
Comment (sofiya02) [inline documents/payment.md:563]: duplicate please here

File: documents/payment.md [hunk 3]
```diff
@@ -466,6 +466,191 @@ body, err := client.Payment.CreatePaymentJson(para_attr, nil)
...
+-------------------------------------------------------------------------------------------------------
+
+### Valid VPA (Third party validation)
+
+```go
+para_attr := map[string]interface{}{
+  "vpa": "gauravkumar@exampleupi",
+}
+
+body, err := client.Payment.ValidateVpa(para_attr, nil)
```
Comment (sofiya02) [inline documents/payment.md:621]: client.Payment.ValidateVpa undefined

**Review by sofiya02 — CHANGES_REQUESTED** [review]

File: documents/payment.md (hunk 3 above)
Comment (ankitdas13) [reply documents/payment.md:621]: try with this 
`para_attr := map[string]interface{}{
     "vpa": "success@razorpay",
 }`
	  
`body, err := client.Payment.ValidateVpa(para_attr, nil)`

File: documents/payment.md [hunk 4]
```diff
@@ -466,6 +466,191 @@ body, err := client.Payment.CreatePaymentJson(para_attr, nil)
...
+}
+```
+-------------------------------------------------------------------------------------------------------
+
+### Fetch payment methods (Third party validation)
+
+```go
+client := razorpay.NewClient("key", "")  // Use Only razorpay key
+
+body, err := client.Payment.FetchMethods(nil, nil)
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply documents/payment.md:646]: i have tested this it's working can you please check the alignment

File: documents/payment.md [hunk 5]
```diff
@@ -466,6 +466,191 @@ body, err := client.Payment.CreatePaymentJson(para_attr, nil)
...
+  "notes": map[string]interface{}{
+    "note_key": "value1",
+  },
+  "upi": map[string]interface{}{
+    "flow": "collect",
+    "vpa": "gauravkumar@exampleupi",
+    "expiry_time": 5,
+  },
+}
+body, err := client.Payment.CreateUpi(para_attr, nil)
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply documents/payment.md:535]: these are order id's for testing 
`order_JpfOO6GYFrQykn, order_JpfOdEIa0dtJPW, order_JpfOkd2OOPAYTu, order_JpfOrcCCePYXlG`
try with this

`para_attr := map[string]interface{}{
		"amount": 100,
		"currency": "INR",
		"order_id": "order_JpfNmveJyTfyDj",
		"email": "gaurav.kumar@example.com",
		"contact": "9090909090",
		"method": "upi",
		"ip": "192.168.0.103",
		"referer": "http",
		"user_agent": "Mozilla/5.0",
		"description": "Test flow",
		"notes": map[string]interface{}{
		  "purpose": "UPI test payment",
		},
		"upi": map[string]interface{}{
		  "flow" : "intent",
		},
	}`
	
`body, err := client.Payment.CreateUpi(para_attr, nil)`

File: documents/payment.md (hunk 2 above)
Comment (ankitdas13) [reply documents/payment.md]: Differentiated 
[Create payment using vpa token](https://github.com/razorpay/razorpay-go/blob/third_party_validation_payment/documents/payment.md?plain=1#L510)

[Create payment upi s2s ](https://github.com/razorpay/razorpay-go/blob/third_party_validation_payment/documents/payment.md?plain=1#L563)

**Pull request merged**